pkg/proxy/apis/config/validation/validation_test.go

```diff
@@ -447,295 +419,286 @@ func TestValidateKubeProxyConfiguration(t *testing.T) {
-		{
+		"SyncPeriod == MinSyncPeriod": {
 			config: kubeproxyconfig.KubeProxyIPVSConfiguration{
 				SyncPeriod:    metav1.Duration{Duration: 10 * time.Second},
 				MinSyncPeriod: metav1.Duration{Duration: 10 * time.Second},
 			},
-			expectErr: false,
+			expectedErrs: field.ErrorList{},
 		},
-		// SyncPeriod > MinSyncPeriod
-		{
+		"SyncPeriod should be > MinSyncPeriod": {
 			config: kubeproxyconfig.KubeProxyIPVSConfiguration{
 				SyncPeriod:    metav1.Duration{Duration: 10 * time.Second},
 				MinSyncPeriod: metav1.Duration{Duration: 5 * time.Second},
 			},
-			expectErr: false,
+			expectedErrs: field.ErrorList{},
 		},
-		// SyncPeriod can be 0
-		{
+		"SyncPeriod can be 0": {
 			config: kubeproxyconfig.KubeProxyIPVSConfiguration{
 				SyncPeriod:    metav1.Duration{Duration: 5 * time.Second},
 				MinSyncPeriod: metav1.Duration{Duration: 0 * time.Second},
 			},
-			expectErr: false,
+			expectedErrs: field.ErrorList{},
 		},
-		// IPVS Timeout can be 0
-		{
+		"IPVS Timeout can be 0": {
 			config: kubeproxyconfig.KubeProxyIPVSConfiguration{
 				SyncPeriod:    metav1.Duration{Duration: 5 * time.Second},
 				TCPTimeout:    metav1.Duration{Duration: 0 * time.Second},
 				TCPFinTimeout: metav1.Duration{Duration: 0 * time.Second},
 				UDPTimeout:    metav1.Duration{Duration: 0 * time.Second},
 			},
-			expectErr: false,
+			expectedErrs: field.ErrorList{},
 		},
-		// IPVS Timeout > 0
-		{
+		"IPVS Timeout > 0": {
 			config: kubeproxyconfig.KubeProxyIPVSConfiguration{
 				SyncPeriod:    metav1.Duration{Duration: 5 * time.Second},
 				TCPTimeout:    metav1.Duration{Duration: 1 * time.Second},
 				TCPFinTimeout: metav1.Duration{Duration: 2 * time.Second},
 				UDPTimeout:    metav1.Duration{Duration: 3 * time.Second},
 			},
-			expectErr: false,
+			expectedErrs: field.ErrorList{},
 		},
-		// TCPTimeout Timeout < 0
-		{
+		"TCPTimeout Timeout < 0": {
 			config: kubeproxyconfig.KubeProxyIPVSConfiguration{
 				SyncPeriod: metav1.Duration{Duration: 5 * time.Second},
 				TCPTimeout: metav1.Duration{Duration: -1 * time.Second},
 			},
-			expectErr: true,
-			reason:    "TCPTimeout must be greater than or equal to 0",
+			expectedErrs: field.ErrorList{field.Invalid(newPath.Child("KubeIPVSConfiguration.TCPTimeout"), metav1.Duration{Duration: -1 * time.Second}, "must be greater than or equal to 0")},
 		},
-		// TCPFinTimeout Timeout < 0
-		{
+		"TCPFinTimeout Timeout < 0": {
 			config: kubeproxyconfig.KubeProxyIPVSConfiguration{
 				SyncPeriod:    metav1.Duration{Duration: 5 * time.Second},
 				TCPFinTimeout: metav1.Duration{Duration: -1 * time.Second},
 			},
-			expectErr: true,
-			reason:    "TCPFinTimeout must be greater than or equal to 0",
+			expectedErrs: field.ErrorList{field.Invalid(newPath.Child("KubeIPVSConfiguration.TCPFinTimeout"), metav1.Duration{Duration: -1 * time.Second}, "must be greater than or equal to 0")},
 		},
-		// UDPTimeout Timeout < 0
-		{
+		"UDPTimeout Timeout < 0": {
```

Comment:
since you can test for multiple errors at once now you could just merge all three "< 0" tests into a single test case couldn't you?

Reply:
Yes that sounds great